lib/internal/Magento/Framework/Mail/Template/TransportBuilder.php

```diff
@@ -277,4 +278,29 @@ protected function prepareMessage()
 
         return $this;
     }
+
+    /**
+     * Add a file joined to the email
+     *
+     * @param string $directory is the directory the file is located
+     * @param string $fileName is the name of the file to join
+     * @param string $fileType is the type of the file, eg: application/pdf
+     * @param string $joinedFileName is the name given to the joined file
+     * @throws \Magento\Framework\Exception\FileSystemException
+     */
+    public function joinFile($directory, $fileName, $fileType, $joinedFileName)
```

Comment:
You almost made my head explode right there.
I think you've meant to write "attachFile", haven't you?